### ISSUE TITLE
Docker buildx to linux/amd64 from Mac M1 throws "system lacks support for the sse3 instruction set"

### ISSUE THREAD

**Issue by bruno-uy** [assafelovic/gpt-researcher#658]
I'm on a Mac M1 and I need to build the Docker image using the following command:
```
docker buildx build --platform linux/amd64 -t image-name-x .
```

That throws this error (I'm only copying the last part):
```
123.1 Setting up notification-daemon (3.20.0-4) ...
123.2 Setting up chromium (120.0.6099.224-1~deb11u1) ...
123.2 update-alternatives: using /usr/bin/chromium to provide /usr/bin/x-www-browser (x-www-browser) in auto mode
123.3 update-alternatives: using /usr/bin/chromium to provide /usr/bin/gnome-www-browser (gnome-www-browser) in auto mode
123.3 Setting up system-config-printer-common (1.5.14-1) ...
123.9 Setting up chromium-driver (120.0.6099.224-1~deb11u1) ...
123.9 Setting up termit (3.1-1) ...
124.0 update-alternatives: using /usr/bin/termit to provide /usr/bin/x-terminal-emulator (x-terminal-emulator) in auto mode
124.0 update-alternatives: warning: skip creation of /usr/share/man/man1/x-terminal-emulator.1.gz because associated file /usr/share/man/man1/termit.1.gz (of link group x-terminal-emulator) doesn't exist
124.0 Setting up system-config-printer (1.5.14-1) ...
124.5 Setting up liblwp-protocol-https-perl (6.10-1) ...
124.5 Setting up libwww-perl (6.52-1) ...
124.5 Setting up libxml-parser-perl:amd64 (2.46-2) ...
124.5 Setting up libxml-twig-perl (1:3.52-1) ...
124.5 Setting up libnet-dbus-perl (1.2.0-1+b1) ...
124.5 Processing triggers for libc-bin (2.31-13+deb11u6) ...
124.5 Processing triggers for dbus (1.12.28-0+deb11u1) ...
124.6 Processing triggers for libgdk-pixbuf-2.0-0:amd64 (2.42.2+dfsg-1+deb11u2) ...
124.8 The hardware on this system lacks support for the sse3 instruction set.
124.8 The upstream chromium project no longer supports this configuration.
124.8 For more information, please read and possibly provide input to their
124.8 bug tracking system at http://crbug.com/1123353
------
Dockerfile:3
--------------------
   2 |
   3 | >>> RUN apt-get update \
   4 | >>>     && apt-get satisfy -y \
   5 | >>>     "chromium, chromium-driver (>= 115.0)" \
```

I solved that by changing the `chromium` installation section in the Dockerfile with the following:
```dockerfile
RUN apt-get update \
    && apt-get install -y gnupg wget ca-certificates --no-install-recommends \
    && wget -qO - https://dl.google.com/linux/linux_signing_key.pub | apt-key add - \
    && echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" > /etc/apt/sources.list.d/google-chrome.list \
    && apt-get update \
    && apt-get install -y google-chrome-stable chromium-driver \
    && google-chrome --version && chromedriver --version
```

**Comment by 0x11c11e** [comment]
It seems that the problem arises due to the Chromium package requiring the SSE3 instruction set, which is not supported by the ARM architecture on the Mac M1.

**Comment by bruno-uy** [comment]
Yes. I solved that by changing that section. No issues with that.

**Comment by pratikpparikh** [comment]
@assafelovic what did you do to solve this?

**Comment by frankcaoyun** [comment]
May I know why this issue is closed without a PR to fix the issue? There are other users using mac to build image to be run in a linux machine e.g. a cloud VM.

**Comment by assafelovic** [comment]
@frankcaoyun based on @bruno-uy suggestion seems that it helps with this. Can you verify? Reopening for now.

**Comment by ElishaKay** [comment]
Sup @frankcaoyun 

Sounds reasonable.

Will the `docker compose up --build` command still work out of the box for ubuntu?

Green light for the PR

**Comment by frankcaoyun** [comment]
Hi @ElishaKay , I don't have a ubuntu machine but the `docker compose` command works in a debian-based VM on GCP.

Our challenge is to use a single docker image to productionize the application because we are adopting Container-Optimized OS on GCP, which requires the application to be in a single image and not running with `docker compose`, and to iterate the development faster, the developers would need to build the image locally (regardless of the machine type) that can work with the linux machine, hence this request.

What are the considerations of having a separate nextjs front-end docker service, instead of combining it into a single app docker image? The nextjs front-end works better cause it supports multi-line prompts, file uploads, etc, which I don't see why we need to make it separately.

**Comment by ElishaKay** [comment]
The NextJS container is built on a NodeJS base image.
The GPTR API Server is built on a python base image.

What you can do, if you want to connect the two is:

Add an environment variable to your NextJS container:

```
NEXT_PUBLIC_GPTR_API_URL=https://domain-of-my-gptr-server.com
```

[More on that here](https://docs.gptr.dev/docs/gpt-researcher/frontend/nextjs-frontend)

**Comment by ElishaKay** [comment]
Hey @frankcaoyun 

Another idea: there is a feature in NextJS to export the static html and js to a public folder.
If you can run `npm run build` on the NextJS Service, it will export the static html and js - at that point, you can place those files within the publicly served folder of your python service.

**Comment by ElishaKay** [comment]
Update: I incorporated @bruno-uy's Dockerfile fix here: https://github.com/assafelovic/gpt-researcher/pull/1191

Was only able to test on ubuntu
@frankcaoyun, can you pull that branch and see if it fixes your mac m1 error?

**Comment by ElishaKay** [comment]
[Based on this](https://github.com/assafelovic/gpt-researcher/issues/1201#issuecomment-2682955840) looks like `linux/amd64` should now be supported

Please feel free to re-open if the issue persists

**Comment by ElishaKay** [comment]
> Hi [@ElishaKay](https://github.com/ElishaKay) , I don't have a ubuntu machine but the `docker compose` command works in a debian-based VM on GCP.
> 
> Our challenge is to use a single docker image to productionize the application because we are adopting Container-Optimized OS on GCP, which requires the application to be in a single image and not running with `docker compose`, and to iterate the development faster, the developers would need to build the image locally (regardless of the machine type) that can work with the linux machine, hence this request.
> 
> What are the considerations of having a separate nextjs front-end docker service, instead of combining it into a single app docker image? The nextjs front-end works better cause it supports multi-line prompts, file uploads, etc, which I don't see why we need to make it separately.

@frankcaoyun 

Try the [embed script ](https://github.com/assafelovic/gpt-researcher/pull/1266)

If you paste that full html embed sample into [frontend/index.html](https://github.com/assafelovic/gpt-researcher/blob/master/frontend/index.html)

That should give you the NextJS app on localhost:8000
